models/repo.go

```diff
@@ -634,6 +634,25 @@ func (repo *Repository) CanBeForked() bool {
 	return !repo.IsBare && repo.UnitEnabled(UnitTypeCode)
 }
 
+// CanUserFork returns true if specified user can fork repository.
+func (repo *Repository) CanUserFork(user *User) (bool, error) {
+	if user == nil {
+		return false, nil
+	}
+	if repo.OwnerID != user.ID && !user.HasForkedRepo(repo.ID) {
+		return true, nil
+	}
+	if err := user.GetOrganizations(true); err != nil {
+		return false, err
+	}
+	for _, org := range user.Orgs {
+		if !org.HasForkedRepo(repo.ID) {
```

Comment:
maybe 
```Go
if org.HasForkedRepo(repo.ID) {
    return false
}
}
return true

Reply:
Not possible, since it would stop looping of all left organizations if first has already forked the repository.

Reply:
@lunny No, that would not work correct. We are looking in this code to check if user has at least one organization that has **not** forked repository yet. And if user has forked && all orgs has forks than it is not possible to fork anymore - so return false

Reply:
@lafriks OK. I see.